prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+
+    :param dict sub_by_coord_dict: dict mapping the (x, y) to a substation
+    :param dict sub_name_dict: dict mapping the substation name to a substation
+    :return: (*list*) -- lines, a list of lines after the unambiguous substation name cleanup.
+    :return: (*list*) -- nodes, a list of substations after the unambiguous substation name cleanup.
+    :return: (*dict*) -- N_dict, a dict mapping the substation to its neighbor list.
+    """
+
+    N_dict = {}
+    nodes = []
+    lines = []
+    missinglines = []
+    if not os.path.isfile("data/unzip/Electric_Power_Transmission_Lines.geojson"):
+        with zipfile.ZipFile(
+            "data/Electric_Power_Transmission_Lines.geojson.zip", "r"
+        ) as zip_ref:
+            zip_ref.extractall("data/unzip")
+    with open(
+        "data/unzip/Electric_Power_Transmission_Lines.geojson", "r", encoding="utf8"
+    ) as fp:
+        data = json.load(fp)
+    for i in range(len(data["features"])):
+        line = data["features"][i]
+        line_id = line["properties"]["ID"]
+        line_type = line["properties"]["TYPE"]  # e.g. "AC; OVERHEAD"
+        if (
+            line["properties"]["SUB_1"] == "NOT AVAILABLE"
+            or line["properties"]["SUB_2"] == "NOT AVAILABLE"
+        ):
+            # quite a few of NOT AVAILABLE scenarios in transmission rawdata
+            print("INFO: sub1 or sub2 NOT AVAILABLE for transmission line: ", line)
+            missinglines.append(line)
+            continue
+
+        if (
+            line["properties"]["SUB_1"] not in sub_name_dict
+            or line["properties"]["SUB_2"] not in sub_name_dict
+        ):
+            # e.g., ID 201978 with SUB2 UNKNOWN202219, never shown in substation rawdata
+            print(
+                "INFO: sub1 or sub2 name not identified for transmission line: ", line
+            )
+            missinglines.append(line)
+            continue
+
+        start_coord = (
+            line["geometry"]["coordinates"][0][0][1],
+            line["geometry"]["coordinates"][0][0][0],
+        )
+        end_coord = (
+            line["geometry"]["coordinates"][0][-1][1],
+            line["geometry"]["coordinates"][0][-1][0],
+        )
+        candidate_coordinates = sub_name_dict.get(
+            line["properties"]["SUB_1"]
+        ) + sub_name_dict.get(line["properties"]["SUB_2"])
+
+        sub1 = min(candidate_coordinates, key=lambda p: computeGeoDist(p, start_coord))
+        sub2 = min(candidate_coordinates, key=lambda p: computeGeoDist(p, end_coord))
+
+        dist = computeGeoDist(start_coord, end_coord)
+
+        lines.append(
+            [
+                line_id,
+                line_type,
+                sub_by_coord_dict.get(sub1)[0],
+                sub_by_coord_dict.get(sub1)[1],
+                sub_by_coord_dict.get(sub2)[0],
+                sub_by_coord_dict.get(sub2)[1],
+                dist,
+            ]
+        )
```

Comment:
Returning values as an unlabeled list here makes the code a bit hard to understand, especially once we get to `calculate_reactance_and_rateA`. Could these be a dictionary or dataframe instead?

Reply:
Too many places depend on the return value `lines`, prefer to leave it for now. but I will add more comments to describe each column.

Reply:
I think the fact that many places rely on a particular ordering of unlabeled entries is all the more reason to change it. This seems like a perfect use case for a data frame instead.

Reply:
Ok, will change it to dataframe.